test/e2e/prune_test.go

```diff
@@ -580,4 +586,57 @@ var _ = Describe("Podman prune", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(dirents).To(HaveLen(3))
 	})
+
+	It("podman system prune --build clean up after terminated build", func() {
+		useCustomNetworkDir(podmanTest, tempdir)
+
+		podmanTest.BuildImage(pruneImage, "alpine_notleaker:latest", "false")
+
+		create := podmanTest.Podman([]string{"create", "--name", "test", BB, "sleep", "10000"})
+		create.WaitWithDefaultTimeout()
+		Expect(create).Should(ExitCleanly())
+
+		containerFilePath := filepath.Join(podmanTest.TempDir, "ContainerFile-podman-leaker")
+		err := os.WriteFile(containerFilePath, []byte(longBuildImage), 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		build := podmanTest.Podman([]string{"build", "-f", containerFilePath, "-t", "podmanleaker"})
+		// Build will never finish so let's wait for build to ask for SIGKILL to simulate a failed build that leaves stage containers
+		for {
+			if build.LineInOutputContains("Please use signal 9") {
+				break
+			}
```

Comment:
this is great but maybe add a small sleep for like 10ms or something to not busy poll which also eats a lot of resources.

Second this needs a timeout, if the build process fails and never prints the output this would loop forever causing hard to debug issues in CI.